### PR TITLE
Paywalls components/container components

### PR DESCRIPTION
**Pull request by jamesrb1**

### CODE REVIEW

File: RevenueCatUI/Templates/Components/StackComponentView.swift [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            }
+            .background(backgroundColor)
+        }
+    }
+
+    init(component: PaywallComponent.StackComponent, locale: Locale, configuration: TemplateViewConfiguration) {
+        self.component = component
+        self.locale = locale
+        self.configuration = configuration
+    }
```
Comment (joshdholtz) [inline RevenueCatUI/Templates/Components/StackComponentView.swift:68]: Nit: `init` goes between variables and `body`

File: RevenueCatUI/Templates/Components/TemplateComponentsView.swift [hunk 2]
```diff
@@ -136,8 +136,8 @@ struct ComponentsView: View {
     }
 
     @ViewBuilder
-    func layoutComponents(_ layoutComponents: [PaywallComponent]) -> some View {
-        ForEach(Array(layoutComponents.enumerated()), id: \.offset) { index, item in
+    func layoutComponents(_ layoutComponentsArray: [PaywallComponent]) -> some View {
+        ForEach(Array(layoutComponentsArray.enumerated()), id: \.offset) { index, item in
```
Comment (joshdholtz) [inline RevenueCatUI/Templates/Components/TemplateComponentsView.swift:140]: Ha! That naming was rather confusing 😅

File: RevenueCatUI/Templates/Components/StackComponentView.swift [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                ComponentsView(locale: locale, components: components, configuration: configuration)
+            }
+            .background(backgroundColor)
+        case .horizontal(let verticalAlignment):
+            HStack(alignment: verticalAlignment.stackAlignment,spacing: spacing) {
+                ComponentsView(locale: locale, components: components, configuration: configuration)
+            }
+            .background(backgroundColor)
+        case .zlayer:
+            ZStack {
```
Comment (joshdholtz) [inline RevenueCatUI/Templates/Components/StackComponentView.swift:57]: Alignment on a `ZStack` might be important too 🤔 Especially if you are trying to overlay like text on image and you want it to be on the bottom? 🤷‍♂️

**Review by joshdholtz — COMMENTED** [review]
Looking good so far! 💪

**Comment by jamesrb1** [comment]
These have been addressed and will be part of the main PR. Thank you!

**Pull request closed without merging**